lib/sparql_to_sw_solr/instance_pub_fields.rb

```diff
@@ -41,15 +51,67 @@ def pub_year_solutions
         sparql.query(query)
       end
 
-      def init_year_fields
-        solution_values_for_binding(pub_year_solutions, :pub_year).each do |value|
-          value = value[0, 4]
-          @year_int ||= Stanford::Mods::DateParsing.year_int_from_date_str(value)
-          @year_display_str ||= Stanford::Mods::DateParsing.date_str_for_display(value)
-          @sort_str ||= Stanford::Mods::DateParsing.sortable_year_string_from_date_str(value)
-          break if @year_int && @year_display_str && @sort_str
-        end
+      def imprint_display
+        edition = solution_values_for_binding(edition_solns, :edition).first
+        pub_info = solution_values_for_binding(publication_solns, :pub_info).first
+        concatenate_values(edition, ' - ', concatenate_values(pub_info, '. ', manu_info))
       end
+
+      def manu_info
+        soln = manufacture_solns.first
+        manu_place = soln.manu_place.to_s if soln && soln.bindings.keys.include?(:manu_place)
+        manu_agent = soln.manu_agent.to_s if soln && soln.bindings.keys.include?(:manu_agent)
+        manu_date = soln.manu_date.to_s if soln && soln.bindings.keys.include?(:manu_date)
+        concatenate_values(manu_place, ' : ', concatenate_values(manu_agent, ', ', manu_date))
+      end
+
+      def edition_solns
+        query = "#{BF_NS_DECL}
+          SELECT ?edition
+          WHERE {
+            <#{instance_uri}> a bf:Instance .
+            <#{instance_uri}> bf:editionStatement ?edition
+          }"
+        sparql.query(query)
+      end
+
+      def publication_solns
+        query = "#{BF_NS_DECL}
+          SELECT ?pub_info
+          WHERE {
+            <#{instance_uri}> a bf:Instance .
+            <#{instance_uri}> bf:provisionActivityStatement ?pub_info .
+          }"
+        sparql.query(query)
+      end
+
+      def manufacture_solns
+        # the query below avoids Manufacture info from marc 008 by filter to blank nodes for place
+        #   and filtering out edtf datatype for dates
+        query = "#{BF_NS_DECL}
+          SELECT ?manu_place ?manu_agent ?manu_date
+          WHERE {
+            <#{instance_uri}> a bf:Instance .
+            <#{instance_uri}> bf:provisionActivity ?provision .
+            ?provision a bf:Manufacture .
+
+            OPTIONAL {
+              ?provision bf:place ?place .
+              FILTER isBlank (?place) .
+              ?place rdfs:label ?manu_place .
+            }
```

Comment:
when it's an IRI, we don't want it -- that means it came from the marc 008.  So isBlank gets us the nodes from marc 260 and 264, which is what we want